channels/offline-channel-js/src/OfflineChannel.ts

```diff
@@ -0,0 +1,604 @@
+                } catch (e) {
+                    // eslint-disable-next-line no-empty
+                }
+                return null;
+                
+            }
+
+            function _constructPayloadData(payloadArr: string[], criticalCnt?: number) {
+
+                try {
+                    if (!_offineSupport) {
+                        return null;
+                    }
+                    let cnt = criticalCnt || 0;
+                    let payload = _offineSupport.batch(payloadArr);
+                    let payloadData: IPayloadData = {
+                        urlString: _urlCfg.url,
+                        data: payload,
+                        headers: {},
+                        criticalCnt: cnt
+                    } as IStorageTelemetryItem;
+                    let details = _offineSupport.getOfflineRequestDetails && _offineSupport.getOfflineRequestDetails();
+                    if (details) {
+                        let url = details.url;
+                        if (url) {
+                            payloadData.urlString = url;
+                        }
+                        let hdrs = details.hdrs;
+                        if (hdrs) {
+                            payloadData.headers = hdrs;
+                        }
+                    }
+                    return payloadData;
+
+                } catch(e) {
+                    // eslint-disable-next-line no-empty
+                }
+                return null;
+            }
+
+            function _createUrlConfig(coreConfig: IConfiguration & IConfig, core: IAppInsightsCore, extensions: IPlugin[], pluginChain?: ITelemetryPluginChain) {
+
+                _self._addHook(onConfigChange(coreConfig, (details) => {
+                    let storageConfig: ILocalStorageConfiguration = null;
+                    let theConfig = details.cfg;
+
+                    let ctx = createProcessTelemetryContext(null, theConfig, core);
+                    storageConfig = ctx.getExtCfg<ILocalStorageConfiguration>(_self.identifier, defaultLocalStorageConfig);
+                    let channelIds = storageConfig.primaryOnlineChannelId;
+                    if (channelIds && channelIds.length) {
+                        arrForEach(channelIds, (id) => {
+                            let plugin = _self.core.getPlugin<IChannelControls>(id);
+                            let channel = plugin && plugin.plugin;
+                            if (channel) {
+                                _primaryChannelId = id;
+                                _offineSupport = channel.getOfflineSupport();
+                                return;
+                            }
+                            
+                        })
+                    }
+                
+                    
+    
+                    let urlConfig: IUrlLocalStorageConfig = _urlCfg;
+                    let curUrl = coreConfig.endpointUrl || DEFAULT_BREEZE_ENDPOINT + DEFAULT_BREEZE_PATH;
```

Comment:
Option: maybe we get the endpoint from the `_offlineSupport` object instead?? and not have any default or `_urlCfg`???

Potential issue however, is that we have an upcoming request to "handle" a Promise for the connection string, so we might not have resolved the endpoint and if it changes it could be problematic) -- although because the `channel.getOfflineSupport()` is inside of an `onConfigChange` the caller could support returning an updated value if it's referenced.